src/lib_ccx/ccx_encoders_python.c

```diff
@@ -3,16 +3,70 @@
 #include "ccx_encoders_common.h"
 #include "ccx_encoders_helpers.h"
 
-#ifdef ENABLE_PYTHON
+#ifdef PYTHON_API
+void Asprintf(char **strp, const char *fmt, ...) {
+	int ret;
+	va_list ap;
+
+	va_start(ap, fmt);
+	ret = vasprintf(strp, fmt, ap);
+	va_end(ap);
+
+	if (ret == -1) {
+		printf("Error: Some problem with asprintf return value in extractor.c\nExiting.");
+		exit(500);
```

Comment:
Should be CCX_COMMON_EXIT_BUG_BUG (a #define in ccx_common_common_h) rather than a magic number.
But do we need to define this function in the first place?

Reply:
Yes, CCX_COMMON_EXIT_BUG_BUG should be better. In the first place? do you mean put the error check handle in the raw asprintf function which locate in utilify.c then we move Asprintf func?